src/main/java/traincraft/blocks/assemblytables/TileAssemblyTable.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Traincraft
+ * Copyright (c) 2011-2020.
+ *
+ * This file ("TileAssemblyTable.java") is part of the Traincraft mod for Minecraft.
+ * It is created by all people that are listed with @author below.
+ * It is distributed under LGPL-v3.0.
+ * You can find the source code at https://github.com/Traincraft/Traincraft
+ */
+package traincraft.blocks.assemblytables;
+
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.InventoryBasic;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.ItemStackHandler;
+import traincraft.tile.BaseTile;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * This is the Assembly Table tileEntity for all tiers of assembly table. It gets the tier based on where the player is looking.
+ *
+ * @implNote If we can figure out how to register all the assembly table tile entities as one, we don't need the three extra files as wrappers for this.
+ *
+ * @author PseudonymPatel
+ * @since 2020-8-7
+ */
+public class TileAssemblyTable extends BaseTile {
+
+    private final int tier;
+    private GuiAssemblyTable guiAssemblyTable;
+    private ContainerAssemblyTable containerAssemblyTable;
+    
+    //These will be the 10 slots that the train parts will sit in to craft the trains
+    private final AssemblyCraftingItemHandler craftingInventory = new AssemblyCraftingItemHandler(10, this);
+    
+    //These 8 slots are for the small chest-like inventory inside of the train workbench.
+    private final ItemStackHandler storageInventory = new ItemStackHandler(8) {
+        @Override
+        protected void onContentsChanged(int slot) {
+            markDirty();
+        }
+    };
+    
+    //This may not be used, but if we need to store the output for some reason.
+    private final InventoryBasic outputInventory = new InventoryBasic("ASM table output", false, 8);
```

Comment:
please make this into one inventory, otherwise the capability would be screwed. Normally a tile entity holds and inventory that is the size of the amount of slots in the container.

Reply:
I don't think this should be merged. Normally, a tile entity has only one purpose, and therefore only needs a single inventory.  Here, there is a storage/chest-like part, and then the crafting part. I think that combining the two would only confuse the users and mods that will use the capability. Normally, I'd agree with you, but storage and crafting provide totally different purposes. I exposed the storage inventory, so it can work like a chest because I feel like that would have a lot more use to the players than filling the crafting area with items.

However, I could combine the output into the craftingInventory with some rework.

Reply:
Yeah there should be only one inventory for all possible slots.
Instead of ItemStackHandler you can use "traincraft.api.InventorySpecific", because why do you need ItemStackHandler in the first place? For the Container Slots you then use "traincraft.api.SlotInventory" instead of "SlotItemHandler"